### DISCUSSION TITLE
JS Doc to Docusauros

### DISCUSSION THREAD

**Discussion by 201flaviosilva** [facebook/docusaurus#7538] [Q&A] [answered]
Hey there,

I'm developing a [simple canvas rendering library](https://github.com/201flaviosilva/Impacto), currently I've documented practically all the code, based on [JS Doc](https://jsdoc.app/).

But currently, I'm having difficulties in merging the documentation generated by JS Doc with the docusauros documentation.

Does anyone know how to do this magic? 😅

**Comment by Josh-Cena** [comment]
Hey, you might be interested in [TypeDoc](http://typedoc.org/)?

You can also play around with [docusaurus-plugin-typedoc](https://github.com/tgreyuk/typedoc-plugin-markdown/tree/master/packages/docusaurus-plugin-typedoc)

**Reply by 201flaviosilva** [reply]
No, my code in pure JavaScript, so I don’t think this will work for me, but thanks, any way 😁

**Reply by Josh-Cena** [reply]
TypeDoc works on JS files as well (iirc), as long as there's JSDoc. It's just you don't get type documentation, only the prose.

**Reply by 201flaviosilva** [reply]
Oh I didn't know, how do I make this work? 😅

**Comment by 201flaviosilva** [comment, marked as answer]
Here is my plugin solution: https://github.com/201flaviosilva/Impacto/blob/e575ac973452af4468a56279ee3d0a797d96e9fa/docs/plugins/jsdoc2docusaurus.js

And my docusaurus.config.js: https://github.com/201flaviosilva/Impacto/blob/e575ac973452af4468a56279ee3d0a797d96e9fa/docs/docusaurus.config.js

**Comment by jtbandes** [comment]
In case anyone else is looking for more solutions, here are some I found in a quick search:

- https://gist.github.com/slorber/0bf8c8c8001505f0f99a062ac55bf442
- https://github.com/naver/jsdoc-to-mdx
- https://typedoc-plugin-markdown.org/plugins/docusaurus/quick-start
